src/common.rs

```diff
@@ -221,13 +221,13 @@ impl TupleProof {
 /// First we evaluate each poly, then we subtract each s * G
 pub struct CheckPrivateShares {
     /// number of keys
-    n: u32,
+    n: usize,
     /// threshold, where the degree of each poly is (t-1)
-    t: u32,
+    t: usize,
```

Comment:
Curious, `usize` instead of `u64` or `u32`? Same question for `neg_shares`.

Reply:
The goal was to get rid of the conversions inside the C callback, since we can’t return errors from there.

Reply:
Oh wow the `get_point` still has an `as` cast 😮

Reply:
Ahhh I see. Got it.

We use `usize` because we often feed things into functions in the `p256k1` library. That library wraps the libsecp256k1 C library via the rust-bindgen crate. And bindgen assumes `size_t` is `usize` (which is the default). We can change that default but keeping things as is allows `p256k1` to be used on rust targets where the pointer size is not 64-bits.

Reply:
Yeah we need to do another pass on usize vs explicitly sized types.  Anything that doesn’t go over the network is probably better s as usize.